tools/src/prebuilds/ExpoModulesCorePrebuilder.ts

```diff
@@ -0,0 +1,175 @@
+import assert from 'assert';
+import fs from 'fs-extra';
+import path from 'path';
+
+import { podInstallAsync } from '../CocoaPods';
+import { getExpoRepositoryRootDir } from '../Directories';
+import logger from '../Logger';
+import { Package } from '../Packages';
+import * as XcodeGen from './XcodeGen';
+import { ProjectSpec } from './XcodeGen.types';
+import XcodeProject, {
+  flavorToFrameworkPath,
+  spreadArgs,
+  SHARED_DERIVED_DATA_DIR,
+} from './XcodeProject';
+import { Flavor, Framework, XcodebuildSettings } from './XcodeProject.types';
+
+// Generates working files out from expo-modules-core folder and prevents CocoaPods generates unnecessary files inside expo-modules-core folder
+const OUT_OF_TREE_WORKING_DIR = path.join(getExpoRepositoryRootDir(), 'prebuild-ExpoModulesCore');
+const MODULEMAP_FILE = 'ExpoModulesCore.modulemap';
+const UMBRELLA_HEADER = 'ExpoModulesCore-umbrella.h';
+
+export function isExpoModulesCore(pkg: Package) {
+  return pkg.packageName === 'expo-modules-core';
+}
+
+export async function generateXcodeProjectAsync(dir: string, spec: ProjectSpec): Promise<string> {
+  const workingDir = OUT_OF_TREE_WORKING_DIR;
+  await fs.ensureDir(workingDir);
+  logger.log(`   Prebuilding expo-modules-core from ${workingDir}`);
+
+  // Links to expo-modules-core source
+  if (typeof spec.targets?.['ExpoModulesCore'].sources?.[0].path === 'string') {
+    spec.targets['ExpoModulesCore'].sources[0].path = path.join(
+      getExpoRepositoryRootDir(),
```

Comment:
As it's a constant value, we usually use `EXPO_DIR` defined in `Constants.ts`, but it's up to you.